syntax = "proto3";

package dcsm;

import "google/protobuf/timestamp.proto";

// KEM (Kontekstualizirovannaya Edinitsa Pamyati)
// Контекстуализированная Единица Памяти
message KEM {
  string id = 1; // Уникальный идентификатор КЕП
  string content_type = 2; // MIME-тип содержимого, например, "text/plain", "application/json", "image/jpeg"
  bytes content = 3; // Основное содержимое КЕП (текст, сериализованные данные, бинарные данные)
  repeated float embeddings = 4; // Векторные представления (эмбеддинги) содержимого

  map<string, string> metadata = 5; // Гибкие метаданные КЕП:
                                    // "topic": "тема КЕП"
                                    // "source": "источник информации"
                                    // "relevance_score": "оценка релевантности (0.0-1.0)"
                                    // "access_rights": "права доступа (например, 'user_id:rw,group_id:r')"
                                    // "custom_key": "любое другое пользовательское значение"

  google.protobuf.Timestamp created_at = 6; // Время создания КЕП
  google.protobuf.Timestamp updated_at = 7; // Время последнего обновления КЕП
}
